pdr_backend/lake/table.py

```diff
@@ -45,32 +46,14 @@ def is_etl_table(table_name: str) -> bool:
 
 
 @enforce_types
-def drop_tables_from_st(pds: PersistentDataStore, type_filter: str, st):
+def drop_tables_from_st(
+    pds: PersistentDataStore, st, table_names_param: Optional[list[str]] = None
+):
     trunc_count = table_count = 0
-    if type_filter not in ["raw", "etl"]:
-        return
-
-    table_names = pds.get_table_names()
+    table_names = table_names_param if table_names_param else pds.get_table_names()
 
     for table_name in table_names:
-        if type_filter == "etl" and not is_etl_table(table_name):
-            logger.info(
-                "[while dropping etl tables] skipping non-etl table %s", table_name
-            )
-            continue
-
-        if type_filter == "raw" and is_etl_table(table_name):
-            logger.info(
-                "[while dropping raw tables] skipping non-raw table %s", table_name
-            )
-            continue
-
-        logger.info(
-            "[while dropping %s tables] drop table %s starting at %s",
-            type_filter,
-            table_name,
-            st,
-        )
```

Comment:
this looks like it was done while "simplifying" things, but we did this to be explicit right now... we'll change this later after this is all completed